app/assets/javascripts/analytics.js.erb

```diff
@@ -1,7 +1,21 @@
 // Start cookie banner (display settings controlled internally by cookies)
 var element = document.querySelector('[data-module="cookie-banner"]')
+var REFERENCE_NUMBER_PATTERN = /\d{8}-\d{6}-.{6}/
+
 new GOVUK.Modules.CookieBanner().start($(element))
 
+var stripPII = function(url) {
+  url = url.replace(REFERENCE_NUMBER_PATTERN, 'reference_number')
+  url = url.replace('true', 'contact_gp')
+  url = url.replace('false', 'contact_gp')
+
+  if(url.includes('reference_number') || 'contact_gp') {
+    return url
+  } else {
+    return location.protocol + '//' + location.host + location.pathname
+  }
```

Comment:
I don't think I understand why this `if/else` is needed. What happens if the function always returns `url`?

Reply:
This was a result of the conversation [here](https://github.com/alphagov/govuk-coronavirus-business-volunteer-form/pull/361). Originally I just returned `url` but using `if/else` was cautionary incase other params are added and we miss changing this code. Have realised its still a bit flawed in that if only one of `reference_number` or `contact_gp` are matched then the other is not redacted, will change to an `&&`...

Reply:
I didn't realise that we're trying to avoid sending other params that could be in the query-string to GA too.  Do you think it's possible to loop through all/any of the query params and replace them all with "XXXXXXX"?

Reply:
I've fallen down a rabbit hole. Apparently there's a `URLSearchParams` class that gets query params and interrogates query params: https://flaviocopes.com/urlsearchparams/

Reply:
nice! Thanks, I will try that :)

Reply:
@leenagupte thanks for suggested, is a much stronger approach. I have amended the commit.